pkg/action/executor/k8s_controller.go

```diff
@@ -149,10 +150,58 @@ func (c *parentController) update(updatedSpec *k8sControllerSpec) error {
 		err = c.ormClient.Update(origControllerObj, c.obj, ownerInfo)
 	} else {
 		_, err = c.clients.dynNamespacedClient.Update(context.TODO(), c.obj, metav1.UpdateOptions{})
+		if utilfeature.DefaultFeatureGate.Enabled(features.AllowIncreaseNsQuota4Resizing) {
+			isReady, err4Waiting := c.waitForReplicasToBeScheduled(c.obj.GetName())
+			if !isReady {
+				glog.V(2).Infof("Timeout on waiting for new replicasets to be scheduled for the workload controller %s/%s", c.obj.GetNamespace(), c.obj.GetName())
+				return err4Waiting
+			}
+			glog.V(2).Infof("All of the new replicasets get scheduled for the workload controller %s/%s", c.obj.GetNamespace(), c.obj.GetName())
+		}
 	}
 	return err
 }
 
+func (c *parentController) waitForReplicasToBeScheduled(name string) (bool, error) {
+	if err := wait.Poll(DefaultRetrySleepInterval, DefaultWaitReplicaToBeScheduled, func() (bool, error) {
+		obj, errInternal := c.clients.dynNamespacedClient.Get(context.TODO(), name, metav1.GetOptions{})
+		if errInternal != nil {
+			return false, errInternal
+		}
+
+		var replicas, updatedReplicas int64
+		found := false
+
+		if obj.GetKind() == util.KindDaemonSet {
+			replicas, found, errInternal = unstructured.NestedInt64(obj.Object, "status", "desiredNumberScheduled")
+		} else {
+			replicas, found, errInternal = unstructured.NestedInt64(obj.Object, "spec", "replicas")
+		}
+		if errInternal != nil || !found {
+			return false, errInternal
+		}
+
+		if obj.GetKind() == util.KindDaemonSet {
+			updatedReplicas, found, errInternal = unstructured.NestedInt64(obj.Object, "status", "updatedNumberScheduled")
+		} else {
+			updatedReplicas, found, errInternal = unstructured.NestedInt64(obj.Object, "status", "updatedReplicas")
+		}
+		if errInternal != nil || !found {
+			return false, errInternal
+		}
+
+		if replicas == updatedReplicas {
+			return true, nil
+		}
+
+		return false, nil
+	}); err != nil {
```

Comment:
you dont need lines 198 to 202. You can simply return err from retry function because you are basing your condition check on that.

Reply:
> you dont need lines 198 to 202

Reply:
> you dont need lines 198 to 202. You can simply return err from retry function because you are basing your condition check on that.

Have deleted these lines

Reply:
> Additionally, as talked earlier, in case of the timeout (failure), I would recommend that the events on this workload are captured and if any of them is type `EventTypeWarning` its packed with the error information and sent back with action failure msg. Please take reference for how to capture events for a given resource from how we do it for pods [here](https://github.com/turbonomic/kubeturbo/blob/master/pkg/discovery/util/pod_util.go#L371). I would not block this work on capturing events. It can be done in a separate task/story.

Add a new function `getLatestWarningEvents` to retrieve the latest events related to the workload controller and package it to the failure message.